### PR TITLE
Move logging functionality into a class

### PR DESCRIPTION
**Pull request by larskraemer**
I moved most of the logging-related code into a class. This eliminates a bunch of global variables and also gives us a single point where we can change the code later.
There is now a global `unique_ptr<Logger>` which handles the logging. I wasn't quite sure where an appropriate entry-point would be, to construct the global, so I added checks in the functions that use it. If anyone who is more familiar with the codebase knows where that would be more appropriate to do, I will gladly change it.

### CODE REVIEW

**Comment by flightlessmango** [comment]
Could probably put the output under `NDEBUG`

**Comment by larskraemer** [comment]
Oh my bad, I was debugging something and forgot to take it out. I'll keep the one in the constructor for future troubleshooting and remove the rest.

**Comment by larskraemer** [comment]
Alright, I cleaned up my code a bit. As far as I'm concerned, this is done for now.

File: src/overlay.cpp [hunk 1]
```diff
@@ -1151,7 +1143,11 @@ void render_benchmark(swapchain_stats& data, struct overlay_params& params, ImVe
 
 void render_imgui(swapchain_stats& data, struct overlay_params& params, ImVec2& window_size, bool is_vulkan)
 {
+<<<<<<< HEAD
```
Comment (jackun) [inline src/overlay.cpp:1146]: Missed this

Comment (larskraemer) [reply src/overlay.cpp]: Thanks, my bad. Fixed now

**Pull request merged**